internal/fs/buffer/writebuffer.go

```diff
@@ -0,0 +1,18 @@
+package buffer
+
+type WriteBuffer interface {
+	// Create creates a buffer of 2*size passed as a parameter.
+	Create(size int) WriteBuffer
+
+	// Write writes to the buffer.
+	Write(data []byte, offset int64) error
+
+	// Upload asynchronously uploads written data to GCS.
+	Upload()
```

Comment:
It should return error at least, and probably some kind of status value/struct.

Reply:
It is hard to predict all the parameters and values that will be required for specific methods. This PR is just for buffer creation. The exact function parameters will be taken care of in their respective separate PRs and would probably require more iterations.